File: MyApplication2/app/src/main/java/com/android/hootr/myapplication/ReactivBT.java
package com.android.hootr.myapplication;

import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;
import io.reactivex.functions.Cancellable;

public class ReactivBT implements ObservableOnSubscribe<DataBT>, Cancellable {

    private ObservableEmitter<DataBT> emitter;

    @Override
    public void subscribe(ObservableEmitter<DataBT> emitter) throws Exception {

        this.emitter = emitter;
        this.emitter.setCancellable(this);
    }

    @Override
    public void cancel() throws Exception {

    }

    public void sendEmitt(DataBT data) {
        if (emitter != null) {
            emitter.onNext(data);
        }
    }
}

File: TSD/app/src/main/java/com/android/hootr/tsd/servises/barcode/ReactivBT.kt
package com.android.hootr.tsd.servises.barcode

import io.reactivex.FlowableEmitter
import io.reactivex.FlowableOnSubscribe

class ReactivBT: FlowableOnSubscribe<DataBT> {

    private var e: FlowableEmitter<DataBT>? = null

    fun sendEmitt(data: DataBT) {

        e?.onNext(data)
    }

    override fun subscribe(emitter: FlowableEmitter<DataBT>) {

        e = emitter

    }
}